java/vector/src/main/java/org/apache/arrow/vector/BaseFixedWidthVector.java

```diff
@@ -499,31 +497,25 @@ public void loadFieldBuffers(ArrowFieldNode fieldNode, List<ArrowBuf> ownBuffers
    */
   public List<ArrowBuf> getFieldBuffers() {
     List<ArrowBuf> result = new ArrayList<>(2);
-    setReaderAndWriterIndex();
-    result.add(validityBuffer);
-    result.add(valueBuffer);
+    result.add(sliceValidityBuffer());
+    result.add(sliceValueBuffer());
 
     return result;
   }
 
-  /**
-   * Set the reader and writer indexes for the inner buffers.
-   */
-  private void setReaderAndWriterIndex() {
-    validityBuffer.readerIndex(0);
-    valueBuffer.readerIndex(0);
+  private ArrowBuf sliceValidityBuffer() {
     if (valueCount == 0) {
-      validityBuffer.writerIndex(0);
-      valueBuffer.writerIndex(0);
-    } else {
-      validityBuffer.writerIndex(getValidityBufferSizeFromCount(valueCount));
-      if (typeWidth == 0) {
-        /* specialized handling for BitVector */
-        valueBuffer.writerIndex(getValidityBufferSizeFromCount(valueCount));
-      } else {
-        valueBuffer.writerIndex(valueCount * typeWidth);
-      }
+      return validityBuffer.slice(0, 0);
+    }
+    return validityBuffer.slice(0, getValidityBufferSizeFromCount(valueCount));
+  }
+
+  private ArrowBuf sliceValueBuffer() {
+    if (valueCount == 0) {
+      return valueBuffer.slice(0, 0);
     }
+    return valueBuffer.slice(0, typeWidth == 0 ?
```

Comment:
we should look at whether we're generating extra object in the case that the slice is the same as the original. In that case, we should probably just return the original rather than generate extra objects.